### ISSUE TITLE
Adding new catalog gives 401

### ISSUE THREAD

**Issue by Toriniasty** [rancher/rancher#20649]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug/Question

**Steps to reproduce (least amount of steps as possible):**
Try to add new catalog, doesn't matter if public or private.

**Result:**
Unexpected HTTP status code 401 from [https://zzz.com/part1uri/reponame.git/index.yaml], expected 200
--

**Other details that may be helpful:**
Same catalog works on our another rancher cluster and I can't pinpoint the reason why it might be happening.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.2.2 && 2.2.3 
- Installation option (single install/HA): HA

**Comment by Toriniasty** [comment]
We are self hosting our own private gitlab repo. I've tried opening up repo and made it public for a sec, oauth token was valid, and also tried with my user credentials which all worked on my laptop.

With the tip to give it a go in rancher pod I've managed to pinpoint the issue and sort it out.
Our deployment didn't populate properly our ca which caused rancher to decline the connection to git repo.

So in theory error is fixed, however if I may suggest some feature/enhancement. Would it be possible to give different error then `Unexpected HTTP status code 401 from [https://zzz.com/part1uri/reponame.git/index.yaml], expected 200`? This was definitely confusing as hell. 

When I've tried to check repo from the container it actually gave:
```sh-4.4# git clone https://zzz.com/part1uri/reponame.git
Cloning into 'reponame'...
fatal: unable to access 'https://zzz.com/part1uri/reponame.git/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none```
